advanced-multi-block.php

```diff
@@ -13,25 +13,29 @@
  * @package CreateBlock
  */
 
-if (! defined('ABSPATH') ) {
-  exit;
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
 }
 
-// Include Composer's autoload file.
-if ( file_exists( plugin_dir_path( __FILE__ ) . 'vendor/autoload.php' ) ) {
-  require_once plugin_dir_path( __FILE__ ) . 'vendor/autoload.php';
-} else {
-  wp_trigger_error( 'Advanced Multi Block Plugin: Composer autoload file not found. Please run `composer install`.', E_USER_ERROR );
-  return;
+// Include our bundled autoload if not loaded globally.
+if ( ! class_exists( Advanced_Multi_Block\Plugin_Paths::class ) && file_exists( __DIR__ . '/vendor/autoload.php' ) ) {
+	require_once __DIR__ . '/vendor/autoload.php';
 }
 
-// Instantiate the classes.
-$advanced_multi_block_classes = array(
-  \Advanced_Multi_Block\Plugin_Paths::class,
-  \Advanced_Multi_Block\Register_Blocks::class,
-  \Advanced_Multi_Block\Enqueues::class,
+if ( ! class_exists( Advanced_Multi_Block\Plugin_Paths::class ) ) {
+	wp_trigger_error( 'Advanced Multi Block Plugin: Composer autoload file not found. Please run `composer install`.', E_USER_ERROR );
+	return;
+}
+
+// Instantiate our modules.
+$advanced_multi_block_modules = array(
+	new Advanced_Multi_Block\Register_Blocks( __DIR__ . '/build' ),
```

Comment:
This approach now avoid registering WP hooks early so that classes can all be instantiated before their `init()` logic is called.